File: nuyhus/BOJ_10845.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> q = new LinkedList<>();
        StringBuilder sb = new StringBuilder();

        while (n-- > 0) {
            st = new StringTokenizer(br.readLine());
            String str = st.nextToken();

            if (str.equals("push")) {
                int l = Integer.parseInt(st.nextToken());
                q.offer(l);
            } else if (str.equals("pop")) {
                if (q.isEmpty()) {
                    sb.append("-1\n");
                } else {
                    sb.append(q.poll()).append("\n");
                }
            } else if (str.equals("size")) {
                sb.append(q.size()).append("\n");
            } else if (str.equals("empty")) {
                sb.append(q.isEmpty() ? "1\n" : "0\n");
            } else if (str.equals("front")) {
                sb.append(q.isEmpty() ? "-1\n" : q.peekFirst() + "\n");
            } else if (str.equals("back")) {
                sb.append(q.isEmpty() ? "-1\n" : q.peekLast() + "\n");
            }
        }

        System.out.print(sb);
    }
}

File: limsumi/BOJ_10845.py
import sys
from collections import deque

n = int(sys.stdin.readline())
ans = deque()

for _ in range(n):
    li = sys.stdin.readline().strip().split()

    if li[0] == "push":
        ans.append(int(li[1]))
    elif li[0] == "pop":
        print(ans.popleft() if ans else -1)
    elif li[0] == "size":
        print(len(ans))
    elif li[0] == "empty":
        print(0 if ans else 1)
    elif li[0] == "front":
        print(ans[0] if ans else -1)
    elif li[0] == "back":
        print(ans[-1] if ans else -1)
